### PR TITLE
Nullabilize MTTP

### PR DESCRIPTION
**Pull request by SUPERCILEX**
Built on top of #91 

@sjwall One of my projects is nearing 100% Kotlin, so I decided to add nullability annotations to all the open-source libraries I'm using. Just in case you didn't know, Kotlin uses nullability annotations to infer the correct type. Cheers! 😄

### CODE REVIEW

File: library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/extras/PromptUtils.java [hunk 1]
```diff
@@ -290,30 +289,27 @@ public static void scale(final PointF origin, final RectF base, final RectF out,
...
      */
-    public static boolean isRtlText(final Layout layout, final Resources resources)
+    public static boolean isRtlText(@Nullable final Layout layout, @NonNull final Resources resources)
     {
         boolean result = false;
         if (layout != null)
         {
             // Treat align opposite as right to left by default
             result = layout.getAlignment() == Layout.Alignment.ALIGN_OPPOSITE;
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.ICE_CREAM_SANDWICH)
```
Comment (SUPERCILEX) [inline library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/extras/PromptUtils.java]: `ICE_CREAM_SANDWICH == 14` which is the min sdk of this lib. Too bad there's not a lint warning for this.

File: library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/extras/focals/RectanglePromptFocal.java [hunk 2]
```diff
@@ -42,9 +45,9 @@
     PointF mBaseBoundsCentre;
     RectF mRippleBounds;
     int mBaseAlpha;
-    float mPadding = 40;
```
Comment (SUPERCILEX) [inline library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/extras/focals/RectanglePromptFocal.java]: It's overridden in the constructor.

File: library/build.gradle [hunk 3]
```diff
@@ -55,6 +55,7 @@ configurations {
 
 dependencies {
     implementation 'com.android.support:support-annotations:' + rootProject.supportLibVersion
+    implementation 'com.android.support:support-compat:' + rootProject.supportLibVersion
```
Comment (sjwall) [inline library/build.gradle]: Comparing the generated library aar file without and with support-compat it looks like with support-compat it increases the library size by 709b and the contents by 4.2kb~. When it is only used for 3 simple methods and trying to keep the library as small as possible it doesn't seem worth while using. 
image

File: library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/extras/PromptUtils.java [hunk 4]
```diff
@@ -203,16 +209,6 @@ else if ((gravity & Gravity.END) == Gravity.END)
         return alignment;
     }
 
-    /**
-     * Determines if Android is after Jelly Bean MR1.
-     *
-     * @return True if running on Android after Jelly Bean MR1.
-     */
-    public static boolean isVersionAfterJellyBeanMR1()
```
[the remark this answers is not in the document]
Comment (SUPERCILEX) [reply library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/extras/PromptUtils.java]: Thanks for catching that! I didn't notice that it was public while in the zone, but yeah, I do follow semantic versioning. 😄

File: library/build.gradle (hunk 3 above)
Comment (SUPERCILEX) [reply library/build.gradle]: SGTM! I'll copy over the support lib implementations with a comment saying where they came from.

**Comment by SUPERCILEX** [comment]
@sjwall It looks like we have a green build! Is this PR good to merge? 🚀😄

**Comment by sjwall** [comment]
Thanks again for your work!

**Comment by SUPERCILEX** [comment]
No problem! 😁

**Comment by sjwall** [comment]
Released in v2.5.0

**Comment by SUPERCILEX** [comment]
Wow, I'm always amazed at how fast you release new versions! 😄 Thanks!

**Pull request merged**